Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/exposurewindows/AnalyticsExposureWindowCollector.kt

```diff
@@ -0,0 +1,50 @@
+package de.rki.coronawarnapp.datadonation.analytics.modules.exposurewindows
+
+import com.google.android.gms.nearby.exposurenotification.ExposureWindow
+import com.google.android.gms.nearby.exposurenotification.ScanInstance
+import de.rki.coronawarnapp.datadonation.analytics.storage.AnalyticsSettings
+import de.rki.coronawarnapp.risk.result.RiskResult
+import javax.inject.Inject
+
+class AnalyticsExposureWindowCollector @Inject constructor(
+    private val analyticsExposureWindowRepository: AnalyticsExposureWindowRepository,
+    private val analyticsSettings: AnalyticsSettings
+) {
+
+    suspend fun reportRiskResultsPerWindow(riskResultsPerWindow: Map<ExposureWindow, RiskResult>) {
+        if (analyticsSettings.analyticsEnabled.value) {
+            collectAnalyticsData(riskResultsPerWindow)
+        }
+    }
+
+    private suspend fun collectAnalyticsData(riskResultsPerWindow: Map<ExposureWindow, RiskResult>) {
+        riskResultsPerWindow.keys.forEach { window ->
+            riskResultsPerWindow[window]?.let { result ->
+                val analyticsExposureWindow = createAnalyticsExposureWindow(
+                    window,
+                    result
+                )
+                analyticsExposureWindowRepository.addNew(analyticsExposureWindow)
+            }
+        }
```

Comment:
why not like this? can result actually be null?
```suggestion
        riskResultsPerWindow.forEach {
            val analyticsExposureWindow = createAnalyticsExposureWindow(
                it.key,
                it.value
            )
            analyticsExposureWindowRepository.addNew(analyticsExposureWindow)
         }
```

Reply:
Both work. I like that I can name it so there is no confusion which one is the key and which one the value.

Reply:
I replaced it